{
  try {
    assertExpression("${file.name}","");
    fail("Should have thrown an exception");
  }
 catch (  ExpressionIllegalSyntaxException e) {
    assertTrue(e.getMessage().startsWith("Unknown function: file.name at location 0"));
  }
  try {
    assertExpression("hey ${xxx} how are you?","");
    fail("Should have thrown an exception");
  }
 catch (  ExpressionIllegalSyntaxException e) {
    assertTrue(e.getMessage().startsWith("Unknown function: xxx at location 4"));
  }
  try {
    assertExpression("${xxx}","");
    fail("Should have thrown an exception");
  }
 catch (  ExpressionIllegalSyntaxException e) {
    assertTrue(e.getMessage().startsWith("Unknown function: xxx at location 0"));
  }
}
